src/main/java/tick/parser/Parser.java

```diff
@@ -0,0 +1,110 @@
+package tick.parser;
+
+import java.time.DateTimeException;
+import java.time.LocalDate;
+
+import tick.commands.AddCommand;
+import tick.commands.Command;
+import tick.commands.CommandType;
+import tick.commands.DeleteCommand;
+import tick.commands.ExitCommand;
+import tick.commands.FindCommand;
+import tick.commands.IncorrectCommand;
+import tick.commands.ListCommand;
+import tick.commands.MarkCommand;
+import tick.commands.UnmarkCommand;
+import tick.exceptions.TickException;
+import tick.tasks.Deadline;
+import tick.tasks.Event;
+import tick.tasks.ToDo;
+
+/**
+ * Represents a parser that parses user input into commands.
+ */
+public class Parser {
+    /**
+     * Parses the user input into a command.
+     *
+     * @param input The user input.
+     * @return The command that corresponds to the user input.
+     * @throws TickException If the user input is invalid.
+     */
+    public static Command parse(String input) throws TickException {
+        String[] tokens = input.split(" ", 2);
+        try {
+            CommandType commandWord = CommandType.valueOf(tokens[0].toUpperCase());
+            switch (commandWord) {
+            case BYE:
+                return new ExitCommand();
+            case LIST:
+                return new ListCommand();
+            case MARK:
+                return new MarkCommand(Integer.parseInt(tokens[1]) - 1);
+            case UNMARK:
+                return new UnmarkCommand(Integer.parseInt(tokens[1]) - 1);
+            case DELETE:
+                return new DeleteCommand(Integer.parseInt(tokens[1]) - 1);
+            case TODO:
+                return Parser.prepareTodo(tokens[1]);
+            case DEADLINE:
+                return Parser.prepareDeadline(tokens[1]);
+            case EVENT:
+                return Parser.prepareEvent(tokens[1]);
+            case FIND:
+                return Parser.prepareFind(tokens[1]);
+            default:
+                return new IncorrectCommand();
+            }
+        } catch (IllegalArgumentException e) {

```

Comment:
I really like how you separated the two Exceptions here and displayed different messages!